### PR TITLE
Close metadatastores in CachingChunkManager.

### PR DESCRIPTION
**Pull request by mansu**
Close metadata stores in `CachingChunkManager` so we are consistent with the rest of the code base.

### CODE REVIEW

File: kaldb/src/main/java/com/slack/kaldb/chunkManager/CachingChunkManager.java [hunk 1]
```diff
@@ -37,26 +41,26 @@ public CachingChunkManager(
...
+    snapshotMetadataStore = new SnapshotMetadataStore(metadataStore, false);
+    searchMetadataStore = new SearchMetadataStore(metadataStore, false);
+    cacheSlotMetadataStore = new CacheSlotMetadataStore(metadataStore, false);
   }
 
   @Override
   protected void startUp() throws Exception {
     LOG.info("Starting caching chunk manager");
 
-    ReplicaMetadataStore replicaMetadataStore = new ReplicaMetadataStore(metadataStore, false);
```
Comment (bryanlb) [inline kaldb/src/main/java/com/slack/kaldb/chunkManager/CachingChunkManager.java:54]: We should keep these initializations in startup, or move all of the codebase to initializing these in the constructor. All other classes initialize temporary metadata stores in the `startup` (ie RecoveryService, IndexingChunkManager, etc).

The reason these are in the startup is an artifact of when the metadatastore could have been "un-started" when these classes were initialized. This is no-longer the case though, as we've moved the metadata startup prior to service initialization .

File: kaldb/src/main/java/com/slack/kaldb/chunkManager/CachingChunkManager.java [hunk 2]
```diff
@@ -86,6 +90,10 @@ protected void shutDown() throws Exception {
           }
         });
 
+    replicaMetadataStore.close();
```
Comment (bryanlb) [inline kaldb/src/main/java/com/slack/kaldb/chunkManager/CachingChunkManager.java:93]: These should be closed in reverse order to initialization.

**Review by bryanlb — CHANGES_REQUESTED** [review]

File: kaldb/src/main/java/com/slack/kaldb/chunkManager/CachingChunkManager.java (hunk 1 above)
Comment (mansu) [reply kaldb/src/main/java/com/slack/kaldb/chunkManager/CachingChunkManager.java:54]: Leaving them in startup for now.

File: kaldb/src/main/java/com/slack/kaldb/chunkManager/CachingChunkManager.java (hunk 2 above)
Comment (mansu) [reply kaldb/src/main/java/com/slack/kaldb/chunkManager/CachingChunkManager.java:93]: Done.

**Review by bryanlb — APPROVED** [review]

**Pull request merged**